report/frontend-development.md

```diff
@@ -1,18 +1,173 @@
+Additionally, npm enhances the security of our project by providing the npm audit tool. This tool scans the project dependencies for known vulnerabilities and security threats. By running npm audit, we can:
+
+ * Detect security issues in real-time across all installed libraries.
+* Receive detailed reports on the nature of vulnerabilities, their severity, and potential fixes.
+* Apply automated patches for certain vulnerabilities using npm audit fix.
+
+By regularly auditing our dependencies, we ensure that our project remains secure and compliant with best practices, reducing the risk of threats from third-party libraries.
+
+In addition to managing dependencies and automating tasks, npm also enables us to run custom test suites. For instance, by specifying test commands in package.json, such as npm run test, we can execute different testing frameworks or tools to ensure the quality and stability of our code.
+
+By leveraging npm for package management, task automation, and security auditing, we ensure that our frontend development workflow is consistent, efficient, and secure. It simplifies collaboration across the team, as all required dependencies and commands are easily accessible, while safeguarding the project against potential vulnerabilities.
+#### Framework Selection Rationale
+
+The project uses **React** with **TypeScript** as the primary framework and language for the frontend, paired with **Vite** as the build tool. This selection is based on the following considerations:
+
+1. **React**:
+   - A component-based architecture makes it easy to build reusable and scalable UI elements.
+   - A large developer community ensures abundant resources and libraries for solving common challenges.
+   - Seamless integration with state management tools and routing libraries like `react-router-dom`.
+
+2. **TypeScript**:
+   - Provides static type-checking, reducing runtime errors.
+   - Helps enforce stricter coding standards by defining types for props, states, and API responses.
+   - Facilitates better code readability and maintainability.
+
+3. **Vite**:
+   - Offers a fast development server with near-instant hot module replacement (HMR).
+   - Simplifies configuration, allowing developers to focus more on building the application.
+   - Produces optimized builds with faster performance compared to traditional tools like Webpack.
+
+This combination was chosen for its ability to balance development speed, scalability, and code quality, making it an ideal stack for a modern, interactive frontend application.
+
+#### Project Structure
+
+Our project structure is meticulously designed to adhere to React best practices.
+
+##### GitHub Workflow
+
+- **ci.yml**: This file defines the GitHub workflow for building and testing the project before deployment.
```

Comment:
Expand pls, or reference that details come later in the devops part of the report.